openupgrade_scripts/scripts/sale_stock/15.0.1.0/end-migration.py

```diff
@@ -0,0 +1,10 @@
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    all_sale_order_lines = env["sale.order.line"].search(
+        [("product_packaging_id", "!=", False)]
+    )
+    for line in all_sale_order_lines:
+        line._onchange_update_product_packaging_qty()
```

Comment:
naive question : 
- why do you realize this action in an end-migration.py file (and not in a post-migration.py) ?

optimization : 
- in case of huge database, maybe write an SQL request instead of calling orm native function could be better. 
(note : not a blocking point, can be done later by other contributors. This implementation is valid)

Reply:
Well it would be end migration to call all installed modules overrides. 

Plus that onchange calls a few other functions. I think it is really end-migration. It does filter out Falses, so on the bright side it might only be slow for huge databases with lots of sale lines using packages.

Reply:
@gdgellatly Thanks you very much